shared/actions/profile/pgp.js

```diff
@@ -48,90 +49,68 @@ function* _dropPgpSaga(action: ProfileGen.DropPgpPayload): Saga.SagaGenerator<an
   }
 }
 
-// TODO(mm) handle error better
-function* _generatePgpSaga(): Saga.SagaGenerator<any, any> {
-  yield Saga.put(navigateTo([peopleTab, 'profile', 'pgp', 'provideInfo', 'generate']))
-
-  const state = yield* Saga.selectState()
-  const {
-    profile: {pgpInfo},
-  } = state
-  const identities = [pgpInfo.email1, pgpInfo.email2, pgpInfo.email3]
-    .filter(email => !!email)
-    .map(email => ({
-      comment: '',
-      email: email || '',
-      username: pgpInfo.fullName || '',
-    }))
-
-  const generatePgpKeyChanMap: any = RPCTypes.pgpPgpKeyGenDefaultRpcChannelMap(
-    [
-      'keybase.1.pgpUi.keyGenerated',
-      'keybase.1.pgpUi.shouldPushPrivate',
-      'keybase.1.pgpUi.finished',
-      'finished',
-    ],
-    {
-      createUids: {
-        ids: identities,
-        useDefault: false,
-      },
+function generatePgp(state: TypedState) {
+  let canceled = false
```

Comment:
I started looking into plumbing this though to the engine saga code but its a lot larger change. instead we just fork off a thing to listen to the cancel button .if it happens we just mark it canceled and nav away. in the next callback we just cancel the rpc and we're done